contracts/ApeProtocol/ApeAllowanceModule.sol

```diff
@@ -27,29 +27,33 @@ abstract contract ApeAllowanceModule {
 	 * @param _circle Circle ID receiving the allowance
 	 * @param _token Address of token to allocate
 	 * @param _amount Amount to take out at most
-	 * @param _interval Duration of an epoch in seconds
+	 * @param _cooldownInterval Duration of an epoch in seconds
 	 * @param _epochs Amount of epochs to fund. Expected_funded_epochs = _epochs + 1
 	 * @param _intervalStart Unix timestamp fromw hich epoch starts (block.timestamp if 0)
 	 */
 	function setAllowance(
 		bytes32 _circle,
 		address _token,
 		uint256 _amount,
-		uint256 _interval,
+		uint256 _cooldownInterval,
 		uint256 _epochs,
-		uint256 _intervalStart
+		uint256 _intervalStart,
+		bool _instantFund
 		) external {
+		uint256 _now = block.timestamp;
+		if (_intervalStart == 0)
+			_intervalStart = _now;
+		require(_instantFund || _intervalStart >= _now, "Interval start in the past");
 		allowances[msg.sender][_circle][_token] = Allowance({
 			maxAmount: _amount,
-			maxInterval: _interval
+			cooldownInterval: _cooldownInterval
 		});
-
 		currentAllowances[msg.sender][_circle][_token] = CurrentAllowance({
 			debt: 0,
-			intervalStart: _intervalStart == 0 ? block.timestamp : _intervalStart,
+			intervalStart: _instantFund ? _now - _cooldownInterval : _intervalStart,
```

Comment:
I think "instant funding" still works if we drop the `_instantFund` argument and just have this...

```suggestion
			intervalStart: _intervalStart,
```

...because it won't check `elapsedTime > _allowance.cooldownInterval` until we try to tap more than has been allowed

Reply:
i think `test_allowance_monthly` demonstrates this? you have `_instantFund=false, epochs=11` in that test and it correctly reverts after 12 epochs